edx/frontend-app-learner-portal-enterprise
# ENT-3405: allow disabling search for integrated customers

**[Link to Ticket](https://openedx.atlassian.net/browse/ENT-3405)**

**[Backend PR](https://github.com/edx/edx-enterprise/pull/956)**

**Disabled Search:**

![image](https://user-images.githubusercontent.com/58956521/94739578-31e75800-033f-11eb-8e59-e0ba7e6c3d93.png)

**Warning Modal:**

![image](https://user-images.githubusercontent.com/58956521/94739301-c7361c80-033e-11eb-8b8e-2ab18ec510c1.png)

**Description:**
* Hide main menu for integrated customers with LP search enabled set to false
* Add `IntegrationWarningModal` to remind integrated customer learners of their company LMS
* Use cookie to allow learner to dismiss integration warnings for the rest of their session
* Upgrade paragon to use the `renderDefaultCloseButton` Modal prop

**Testing Instructions:**
Setup: 
1. A Third Party Auth (TPA) configuration must be setup in your devstack. (see [How to set up Enterprise SSO for Testing](https://openedx.atlassian.net/wiki/spaces/SOL/pages/131531061/How+to+set+up+Enterprise+SSO+for+Testing))
2. Update your test enterprise customer to use your configured TPA as the identity provider and enable the Learner Portal.
3. Login to the Learner Portal (LP) using your test customer slug and register an account if this is your first time logging in with the given SSO credentials. Navigate to the 'Account' page using the user drop down menu in the upper left corner and record the email address to use in the next step.
4. Associate the learner account email with an active License for a Subscription Plan tied to your test customer. Refresh the LP page and observe the error no longer being displayed.

Testing:
1. Disable the "Enable integrated customer learner portal search" field for your test customer. 
2. Refresh the Learner Portal page and observe how the "Dashboard", "Find a Course", and "Help" options are no longer from the main menu in the header.
3. Enable the "Enable integrated customer learner portal search" field for your test customer. 
4. Refresh the page and observe the integration warning modal appear. Dismiss the warning.
5. Use the "Find a Course" option from the main menu to visit the search page. Observe the warning isn't shown.
6. Delete the `integration-warning-dismissed` cookie and refresh the page. Observe the warning is shown again and dismiss. 
7. Log out and back in. Visit either the Dashboard or Search page and observe the warning shown again for the new session.
